torch/csrc/jit/codegen/cuda/ops/alias.cpp

```diff
@@ -193,6 +194,59 @@ TensorView* unsqueeze(TensorView* x, int dim) {
   return broadcast(x, broadcast_axes);
 }
 
+TensorView* permute(TensorView* x, const std::vector<int64_t>& new2old) {
+  auto inp_domain = TensorDomain::noReductions(x->getMaybeRFactorDomain());
+  std::vector<IterDomain*> out_domain(inp_domain.size());
+
+  auto normalized_new2old =
+      ir_utils::normalizeNew2Old(new2old, inp_domain.size());
+
+  for (const auto i : c10::irange(out_domain.size())) {
+    auto in_id = inp_domain[new2old[i]];
+    out_domain[i] = in_id->cloneWithoutRFactor();
+  }
+
+  TensorView* out_tensor = IrBuilder::create<TensorView>(
+      IrBuilder::create<TensorDomain>(
+          out_domain, std::vector<bool>(out_domain.size(), true)),
+      x->getDataType().value());
+  IrBuilder::create<TransposeOp>(out_tensor, x, normalized_new2old);
+  return out_tensor;
+}
+
+TensorView* transpose(TensorView* x, int64_t dim0, int64_t dim1) {
+  const auto ndims = static_cast<int>(x->domain()->noReductions().size());
+
+  // short-circuit: return original tensorview if less than 2 dimensions
+  if (ndims < 2) {
```

Comment:
Should this be just an error?

Reply:
This is eager-mode behavior. https://pytorch.org/docs/stable/generated/torch.t.html
I'll move the short-circuit after the dimension checks.